src/utils/parsers/BaseParser.ts

```diff
@@ -246,61 +209,121 @@ export abstract class BaseParser {
         return allMatches
     }
 
-    private formatMatch(match: MatchWithRange, file: parse.File, change: parse.Change): VariableMatch {
-        return {
-            name: match.name,
-            fileName: file.to ?? '',
-            line: change.type === 'normal' ? change.ln1 : change.ln,
-            mode: change.type === 'add' ? 'add' : 'remove',
-            ...(match.isUnknown ? { isUnknown: true } : {})
-        }
+    /**
+     * Filter unused lines and build an object containing lines and content
+     */
+    private filterAndReduceLines<T>(lines: ParsedLine[]): { content: string, lines: T[] } {
+        return lines
+            .filter((line) => !line.isComment(this.commentCharacters) && !line.isEmpty())
+            .reduce((acc, line) => {
+                line.range.start = acc.content.length
+
+                acc.content += line.parsedContent
+                line.range.end = acc.content.length - 1
+
+                acc.lines.push(line as unknown as T)
+                
+                return acc
+            }, { content: '', lines: [] } as { content: string, lines: T[] })
     }
 
     /**
-     * Given an array of matches, find the first change object corresponding to the match.
-     * Also verify that the match is associated with at least one add/delete change object.
+     * Get all lines within the range of the match
      */
-    private formatMatches(file: parse.File, matches: MatchWithRange[], { changes }: MultilineChunk) {
-        const results: VariableMatch[] = []
+    private getMatchingLines<T>(match: MatchWithRange, lines: ParsedLine[]): T[] {
```

Comment:
how come this needs a generic type arg if it's just type casting to the provided type at the end? Doesn't that mean there's really only one type here?

Reply:
this handles the result being either an array of `ParsedLine` or an array of `ParsedChangeLine` in the diff case

Reply:
but the actual data being returned doesn't look any different so why allow the type of its return to be overridden like this?

Reply:
the returned object type matches what's passed in, `ParsedChangeLine` has additional properties that are used later

Reply:
but what's passed in is always of type "ParsedLine", and so the return is always an array of "ParsedLine"

Reply:
if you're trying to represent two different call signatures here then you should probably use function overloading 
https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads

Reply:
that's cool, but it doesn't seem to work here, even with:
```
    private getMatchingLines(match: MatchWithRange, lines: ParsedChangeLine[]): ParsedChangeLine[];
    private getMatchingLines(match: MatchWithRange, lines: ParsedLine[]): ParsedLine[] {
```
typescript complains:
`Argument of type 'ParsedLine[]' is not assignable to parameter of type 'ParsedChangeLine[]'`

Reply:
ParsedChangeLine extends ParsedLine, so typescript still considers that a ParsedLine

Reply:
scratch that previous comment — just realized the function signature can't be called and it needs another overload